### PR TITLE
configure strict pytest

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
related: https://github.com/agronholm/anyio/pull/309

File: requirements.txt [hunk 1]
```diff
@@ -20,7 +20,8 @@ twine==3.4.1
 wheel==0.36.2
 
 # Tests & Linting
-anyio==3.1.0
+https://github.com/agronholm/anyio/pull/309
+anyio @ https://github.com/agronholm/anyio/archive/refs/heads/9ae936079d407223a899c635b6cc859a64bc0124.tar.gz
```
Comment (graingert) [inline requirements.txt:24]:
```suggestion
anyio @ https://github.com/agronholm/anyio/archive/9ae936079d407223a899c635b6cc859a64bc0124.tar.gz
```

File: requirements.txt [hunk 2]
```diff
@@ -20,7 +20,8 @@ twine==3.4.1
 wheel==0.36.2
 
 # Tests & Linting
-anyio==3.1.0
+https://github.com/agronholm/anyio/pull/309
```
Comment (graingert) [inline requirements.txt:23]:
```suggestion
# https://github.com/agronholm/anyio/pull/309
```

File: requirements.txt [hunk 3]
```diff
@@ -20,7 +20,8 @@ twine==3.4.1
 wheel==0.36.2
 
 # Tests & Linting
-anyio==3.1.0
+# https://github.com/agronholm/anyio/pull/309
+anyio @ https://github.com/agronholm/anyio/archive/9ae936079d407223a899c635b6cc859a64bc0124.tar.gz
```
Comment (graingert) [inline requirements.txt:24]:
```suggestion
anyio @ git+https://github.com/agronholm/anyio.git@9ae936079d407223a899c635b6cc859a64bc0124
```

File: requirements.txt [hunk 4]
```diff
@@ -20,7 +20,8 @@ twine==3.4.1
 wheel==0.36.2
 
 # Tests & Linting
-anyio==3.1.0
+# https://github.com/agronholm/anyio/pull/309
+anyio @ git+https://github.com/agronholm/anyio.git@9ae936079d407223a899c635b6cc859a64bc0124
```
Comment (tomchristie) [inline requirements.txt:24]: Okay, let's consider this change as being blocked on an `anyio` 3.2 release, I guess?

File: requirements.txt [hunk 5]
```diff
@@ -20,7 +20,8 @@ twine==3.4.1
 wheel==0.36.2
 
 # Tests & Linting
-anyio==3.1.0
+# https://github.com/agronholm/anyio/pull/309
```
Comment (graingert) [inline requirements.txt:23]:
```suggestion
```

File: requirements.txt (hunk 4 above)
Comment (graingert) [inline requirements.txt:24]:
```suggestion
anyio~=3.2
```

**Comment by tomchristie** [comment]
Closing a few PRs here, as outdated by the 0.14 release.
If there's work that ought to be picked up again, then we can discuss and consider opening an updated version.

For this particular PR, I'd suggest we could just start with a PR that turns on strict mode, *without* resolving any issues.

Having the failing case first then helps demo that any changes are necessary and sufficient.

**Pull request closed without merging**